### ISSUE TITLE
pyRevitplus - Crop Image tool not working with 4.7.4

### ISSUE THREAD

**Issue by kenmarcus10** [eirannejad/pyRevit#810]
Hi,

I am having an issue with the Crop Image tool on Pyrevit Plus. Here is the version info: 4.7.4:277:2020.2. 

I follow the steps required for the tool to work in the previous version but I am getting different error messages when I try to run:

**this is one error:** 

```
IronPython Traceback:
Traceback (most recent call last):
 File "C:\Users\Administrator\AppData\Roaming\pyRevit\Extensions\PyRevitPlus.extension\pyRevitPlus.tab\Image.panel\Crop Image.pushbutton\script.py", line 171, in <module>
 File "C:\Users\Administrator\AppData\Roaming\pyRevit\Extensions\PyRevitPlus.extension\pyRevitPlus.tab\Image.panel\Crop Image.pushbutton\script.py", line 79, in crop_image
Exception: Image Source path not found: 

Script Executor Traceback:
IronPython.Runtime.Exceptions.PythonException: Image Source path not found: 
 at Microsoft.Scripting.Interpreter.ThrowInstruction.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.Interpreter.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.LightLambda.Run3[T0,T1,T2,TRet](T0 arg0, T1 arg1, T2 arg2)
 at System.Dynamic.UpdateDelegates.UpdateAndExecute4[T0,T1,T2,T3,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3)
 at Microsoft.Scripting.Interpreter.DynamicInstruction`5.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.Interpreter.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.LightLambda.Run2[T0,T1,TRet](T0 arg0, T1 arg1)
 at IronPython.Compiler.PythonScriptCode.RunWorker(CodeContext ctx)
 at PyRevitLabs.PyRevit.Runtime.IronPythonEngine.Execute(ScriptRuntime& runtime)

**this is a second error:**

[ERROR] Error in Transaction Context: has rolled back. 

IronPython Traceback:
Traceback (most recent call last):
 File "C:\Users\Administrator\AppData\Roaming\pyRevit\Extensions\PyRevitPlus.extension\pyRevitPlus.tab\Image.panel\Crop Image.pushbutton\script.py", line 182, in <module>
Exception: OutOfRange
Parameter name: options.Resolution


Script Executor Traceback:
Autodesk.Revit.Exceptions.ArgumentException: OutOfRange
Parameter name: options.Resolution
 at Microsoft.Scripting.Interpreter.ThrowInstruction.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.Interpreter.HandleException(InterpretedFrame frame, Exception exception)
 at Microsoft.Scripting.Interpreter.Interpreter.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.LightLambda.Run2[T0,T1,TRet](T0 arg0, T1 arg1)
 at IronPython.Compiler.PythonScriptCode.RunWorker(CodeContext ctx)
 at PyRevitLabs.PyRevit.Runtime.IronPythonEngine.Execute(ScriptRuntime& runtime)

**here is a third error:**

IronPython Traceback:
Traceback (most recent call last):
 File "C:\Users\Administrator\AppData\Roaming\pyRevit\Extensions\PyRevitPlus.extension\pyRevitPlus.tab\Image.panel\Crop Image.pushbutton\script.py", line 171, in <module>
 File "C:\Users\Administrator\AppData\Roaming\pyRevit\Extensions\PyRevitPlus.extension\pyRevitPlus.tab\Image.panel\Crop Image.pushbutton\script.py", line 79, in crop_image
Exception: Image Source path not found: 

Script Executor Traceback:
IronPython.Runtime.Exceptions.PythonException: Image Source path not found: 
 at Microsoft.Scripting.Interpreter.ThrowInstruction.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.Interpreter.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.LightLambda.Run3[T0,T1,T2,TRet](T0 arg0, T1 arg1, T2 arg2)
 at System.Dynamic.UpdateDelegates.UpdateAndExecute4[T0,T1,T2,T3,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3)
 at Microsoft.Scripting.Interpreter.DynamicInstruction`5.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.Interpreter.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.LightLambda.Run2[T0,T1,TRet](T0 arg0, T1 arg1)
 at IronPython.Compiler.PythonScriptCode.RunWorker(CodeContext ctx)
 at PyRevitLabs.PyRevit.Runtime.IronPythonEngine.Execute(ScriptRuntime& runtime)
```

I have tried in multiple files, Floor plan, drafting and rendering views. None are working. 

Is it possible that the process has changed?  Draw a filled region. Select the image and filled image, choose the crop image tool?

Also, what is the process to do multiple crops on the same image at once? I have tried to follow the video example but can not replicate it. 

thanks for your help.

love pyRevit. 

ken

**Comment by kenmarcus10** [comment]
forgot to mention I have tried all of the other pyRevitPlus tools and those are working fine. It appears to only be happening with the crop image tool

**Comment by eirannejad** [comment]
Would you mind reporting this to pyRevitPlus repo?

https://github.com/gtalarico/pyrevitplus

Thanks and sorry for the inconvenience

**Comment by kenmarcus10** [comment]
Sure thing. Thanks for getting back to me

Thanks,
Ken
415.979.0678
On Jan 31, 2020, 3:23 PM -0800, Ehsan Iran-Nejad <notifications@github.com>, wrote:

Closed #810<https://github.com/eirannejad/pyRevit/issues/810>.
